### PR TITLE
add sign and verify to transit api

### PR DESCRIPTION
**Pull request by redegade**
Added support for the sign and verify capabilities of the transit engine. Added tests for a few basic functionalities.

### CODE REVIEW

File: tests/util.spec.ts [hunk 1]
```diff
@@ -18,6 +18,6 @@ describe("URL Resolve", () => {
 
     test("URL Resolve fails with invalid URL", () => {
         const resolveWrap = () => resolveURL("ht/example", "/test///", "////foo/baz/lul", "/bar///");
-        expect(resolveWrap).toThrowError(new TypeError("Invalid URL: ht/example/test/foo/baz/lul/bar"));
+        expect(resolveWrap).toThrowError(new TypeError("Invalid URL"));
```
Comment (Lucaber) [inline tests/util.spec.ts:21]: Your tests are currently running with node 12 in ci to ensure compatibility.
The error message was changed in node 16 (thanks log4shell, i guess)
This seams to work in both versions.

```suggestion
        expect(resolveWrap).toThrowError(new RegExp("Invalid URL.*"));
```

File: src/engines/transit.ts [hunk 2]
```diff
@@ -210,6 +218,71 @@ export class TransitVaultClient extends AbstractVaultClient {
         });
     }
 
+    /**
```
Comment (Lucaber) [inline src/engines/transit.ts:221]: run `yarn format`

Comment (redegade) [reply src/engines/transit.ts:221]: Made the changes and squashed the commits. I guess we need a very simple CONTRIBUTING.md page: just a quick list of instructions of what to run, i.e. running "docker-compose up" before integration testing etc. I'd gladly help but wouldn't know where to start.

**Comment by Lucaber** [comment]
Thanks, released in 1.5.0

**Pull request merged**